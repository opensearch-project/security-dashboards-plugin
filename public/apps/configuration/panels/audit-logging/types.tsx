/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

export interface GeneralSettings {
  // layer
  enable_rest?: boolean;
  enable_transport?: boolean;
  disabled_rest_categories?: string[];
  disabled_transport_categories?: string[];

  // attribute
  resolve_bulk_requests?: boolean;
  resolve_indices?: boolean;
  log_request_body?: boolean;
  exclude_sensitive_headers?: boolean;

  // ignore
  ignore_users?: string[];
  ignore_requests?: string[];
}

export interface ComplianceSettings {
  enabled?: boolean;

  // read
  read_metadata_only?: boolean;
  read_ignore_users?: string[];
  read_watched_fields?: {
    [key: string]: string;
  };

  // write
  write_metadata_only?: boolean;
  write_log_diffs?: boolean;
  write_ignore_users?: string[];
  write_watched_indices?: string[];
}

export interface AuditLoggingSettings {
  enabled?: boolean;
  audit?: GeneralSettings;
  compliance?: ComplianceSettings;
}
