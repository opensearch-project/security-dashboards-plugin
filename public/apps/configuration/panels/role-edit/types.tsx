/*
 *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { EuiComboBoxOptionOption } from '@elastic/eui';

export type ComboBoxOptions = EuiComboBoxOptionOption[];
export type FieldLevelSecurityMethod = 'exclude' | 'include';
export interface RoleIndexPermissionStateClass {
  indexPatterns: ComboBoxOptions;
  docLevelSecurity: string;
  fieldLevelSecurityMethod: FieldLevelSecurityMethod;
  fieldLevelSecurityFields: ComboBoxOptions;
  maskedFields: ComboBoxOptions;
  allowedActions: ComboBoxOptions;
}

export enum TenantPermissionType {
  None = '',
  Read = 'r',
  Write = 'w',
  Full = 'rw',
}

export interface RoleTenantPermissionStateClass {
  tenantPatterns: ComboBoxOptions;
  permissionType: TenantPermissionType;
}
