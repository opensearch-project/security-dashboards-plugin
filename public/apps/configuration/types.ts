/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { EuiComboBoxOptionOption } from '@elastic/eui';

export type ComboBoxOptions = EuiComboBoxOptionOption[];

export type FieldLevelSecurityMethod = 'exclude' | 'include';

export enum ResourceType {
  roles = 'roles',
  users = 'users',
  permissions = 'permissions',
  tenants = 'tenants',
  tenantsManageTab = 'tenantsManageTab',
  tenantsConfigureTab = 'tenantsConfigureTab',
  auth = 'auth',
  auditLogging = 'auditLogging',
}

export enum Action {
  view = 'view',
  create = 'create',
  edit = 'edit',
  duplicate = 'duplicate',
  manage = 'manage',
  configure = 'configure',
}

export enum SubAction {
  mapuser = 'mapuser',
}

export enum PageId {
  dashboardId = 'dashboards',
  visualizationId = 'visualize',
}

export enum TenantPermissionType {
  None = '',
  Read = 'Read only',
  ReadWrite = 'Read and Write',
}

export interface RouteItem {
  name: string;
  href: string;
}

export interface DataObject<T> {
  [key: string]: T;
}

export interface ObjectsMessage<T> {
  total: number;
  data: DataObject<T>;
}

export interface RoleIndexPermission {
  index_patterns: string[];
  dls: string;
  fls: string[];
  masked_fields: string[];
  allowed_actions: string[];
}

export interface RoleIndexPermissionView extends RoleIndexPermission {
  id: number;
}

export interface RoleTenantPatterns {
  tenant_patterns: string[];
}

export interface RoleTenantPermission extends RoleTenantPatterns {
  allowed_actions: string[];
}

export interface RoleTenantPermissionView extends RoleTenantPatterns {
  permissionType: string;
}

export interface RoleUpdate {
  cluster_permissions: string[];
  index_permissions: RoleIndexPermission[];
  tenant_permissions: RoleTenantPermission[];
}

export interface RoleDetail extends RoleUpdate {
  reserved: boolean;
}

export interface RoleMappingDetail {
  backend_roles: string[];
  hosts: string[];
  users: string[];
}

export interface TenantUpdate {
  description: string;
}

export interface TenantName {
  tenant: string;
}

export interface Tenant extends TenantUpdate, TenantName {
  reserved: boolean;
  tenantValue: string;
}

export interface TenantSelect extends TenantName {
  username: string;
}

export interface RoleTenantPermissionDetail extends RoleTenantPermissionView, Tenant {}

export interface UserAttributes {
  [key: string]: string;
}

export interface InternalUser {
  attributes: UserAttributes;
  backend_roles: string[];
}

export interface InternalUserUpdate extends InternalUser {
  password?: string;
}

export interface ActionGroupUpdate {
  allowed_actions: string[];
}

export interface ActionGroupItem extends ActionGroupUpdate {
  reserved: boolean;
  type: 'cluster' | 'index' | 'all' | 'kibana' | undefined;
}

export interface ExpandedRowMapInterface {
  [key: string]: React.ReactNode;
}

export interface FormRowDeps {
  headerText: string;
  optional?: boolean;
  headerSubText?: string;
  helpLink?: string;
  helpText?: string;
}
