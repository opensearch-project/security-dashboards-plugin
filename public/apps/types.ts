import { AppMountParameters, CoreStart } from '../../../../src/core/public';
import { AppPluginStartDependencies } from '../types';

export interface AppDependencies {
    coreStart: CoreStart,
    navigation: AppPluginStartDependencies,
    params: AppMountParameters
}
