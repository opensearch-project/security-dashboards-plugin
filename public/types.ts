/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { NavigationPublicPluginStart } from '../../../src/plugins/navigation/public';
import {
  SavedObjectsManagementPluginSetup,
  SavedObjectsManagementPluginStart,
} from '../../../src/plugins/saved_objects_management/public';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface SecurityPluginSetup {}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface SecurityPluginStart {}

export interface SecurityPluginSetupDependencies {
  savedObjectsManagement: SavedObjectsManagementPluginSetup;
}

export interface SecurityPluginStartDependencies {
  navigation: NavigationPublicPluginStart;
  savedObjectsManagement: SavedObjectsManagementPluginStart;
}

export interface AuthInfo {
  user_name: string;
  tenants: {
    [tenant: string]: boolean;
  };
}

export interface DashboardsInfo {
  multitenancy_enabled?: boolean;
  private_tenant_enabled?: boolean;
  default_tenant: string;
}

export interface ClientConfigType {
  readonly_mode: {
    roles: string[];
  };
  ui: {
    basicauth: {
      login: {
        title: string;
        subtitle: string;
        showbrandimage: boolean;
        brandimage: string;
        buttonstyle: string;
      };
    };
    anonymous: {
      login: {
        buttonname: string;
        showbrandimage: boolean;
        brandimage: string;
        buttonstyle: string;
      };
    };
    openid: {
      login: {
        buttonname: string;
        showbrandimage: boolean;
        brandimage: string;
        buttonstyle: string;
      };
    };
    saml: {
      login: {
        buttonname: string;
        showbrandimage: boolean;
        brandimage: string;
        buttonstyle: string;
      };
    };
    autologout: boolean;
    backend_configurable: boolean;
  };
  multitenancy: {
    enable_aggregation_view: boolean;
    enabled: boolean;
    tenants: {
      enable_private: boolean;
      enable_global: boolean;
    };
  };
  auth: {
    type: string | string[];
    anonymous_auth_enabled: boolean;
    logout_url: string;
  };
  clusterPermissions: {
    include: string[];
  };
  indexPermissions: {
    include: string[];
  };
  disabledTransportCategories: {
    exclude: string[];
  };
  disabledRestCategories: {
    exclude: string[];
  };
}
