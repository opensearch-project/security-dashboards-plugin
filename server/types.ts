// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface OpendistroSecurityPluginSetup {}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface OpendistroSecurityPluginStart {}
