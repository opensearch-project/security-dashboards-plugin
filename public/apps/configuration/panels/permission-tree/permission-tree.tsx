/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { EuiTreeView, EuiText } from '@elastic/eui';
import { Node } from '@elastic/eui/src/components/tree_view/tree_view';
import React from 'react';
import { isEmpty } from 'lodash';
import { ActionGroupItem, DataObject } from '../../types';

import './_index.scss';

const MAX_DEPTH = 5;

function buildTreeItem(
  name: string,
  depth: number,
  actionGroups: DataObject<ActionGroupItem>
): Node {
  let children: string[] | null = null;
  if (depth < MAX_DEPTH) children = actionGroups[name]?.allowed_actions;

  return {
    label: name,
    id: name,
    icon: <EuiText size="xs">â€¢</EuiText>,
    children: children?.map((child) => buildTreeItem(child, depth + 1, actionGroups)),
    className: isEmpty(children) ? 'tree-leaf-node' : '',
  };
}

export function PermissionTree(props: {
  permissions: string[];
  actionGroups: DataObject<ActionGroupItem>;
}) {
  return (
    <div className="permission-tree-container">
      <EuiTreeView
        display="compressed"
        aria-label="Permission tree"
        showExpansionArrows
        items={props.permissions.map((permission) =>
          buildTreeItem(permission, 0, props.actionGroups)
        )}
      />
    </div>
  );
}
