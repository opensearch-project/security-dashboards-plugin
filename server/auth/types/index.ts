/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

export { BasicAuthentication } from './basic/basic_auth';
export { JwtAuthentication } from './jwt/jwt_auth';
export { OpenIdAuthentication } from './openid/openid_auth';
export { ProxyAuthentication } from './proxy/proxy_auth';
export { SamlAuthentication } from './saml/saml_auth';
export { MultipleAuthentication } from './multiple/multi_auth';
