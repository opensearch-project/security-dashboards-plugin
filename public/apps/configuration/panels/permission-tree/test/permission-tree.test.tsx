/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { shallow } from 'enzyme';
import React from 'react';
import { ActionGroupItem, DataObject } from '../../../types';
import { PermissionTree } from '../permission-tree';

describe('Permission tree', () => {
  const permissions = ['permission1'];
  const actionGroupItem: DataObject<ActionGroupItem> = {
    permission1: {
      allowed_actions: ['permission11'],
      reserved: true,
      type: 'index',
    },
    permission11: {
      allowed_actions: ['permission12'],
      reserved: true,
      type: 'index',
    },
    permission12: {
      allowed_actions: ['permission13'],
      reserved: true,
      type: 'index',
    },
    permission13: {
      allowed_actions: ['permission14'],
      reserved: true,
      type: 'index',
    },
    permission14: {
      allowed_actions: ['permission15'],
      reserved: true,
      type: 'index',
    },
    permission15: {
      allowed_actions: ['permission16'],
      reserved: true,
      type: 'index',
    },
  };

  it('renders', () => {
    const component = shallow(
      <PermissionTree permissions={permissions} actionGroups={actionGroupItem} />
    );
    expect(component).toMatchSnapshot();
  });
});
