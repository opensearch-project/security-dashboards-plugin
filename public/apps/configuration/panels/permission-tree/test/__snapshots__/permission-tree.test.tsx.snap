// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permission tree renders 1`] = `
<div
  className="permission-tree-container"
>
  <EuiTreeView
    aria-label="Permission tree"
    display="compressed"
    items={
      Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "children": undefined,
                              "className": "tree-leaf-node",
                              "icon": <EuiText
                                size="xs"
                              >
                                •
                              </EuiText>,
                              "id": "permission15",
                              "label": "permission15",
                            },
                          ],
                          "className": "",
                          "icon": <EuiText
                            size="xs"
                          >
                            •
                          </EuiText>,
                          "id": "permission14",
                          "label": "permission14",
                        },
                      ],
                      "className": "",
                      "icon": <EuiText
                        size="xs"
                      >
                        •
                      </EuiText>,
                      "id": "permission13",
                      "label": "permission13",
                    },
                  ],
                  "className": "",
                  "icon": <EuiText
                    size="xs"
                  >
                    •
                  </EuiText>,
                  "id": "permission12",
                  "label": "permission12",
                },
              ],
              "className": "",
              "icon": <EuiText
                size="xs"
              >
                •
              </EuiText>,
              "id": "permission11",
              "label": "permission11",
            },
          ],
          "className": "",
          "icon": <EuiText
            size="xs"
          >
            •
          </EuiText>,
          "id": "permission1",
          "label": "permission1",
        },
      ]
    }
    showExpansionArrows={true}
  />
</div>
`;
